client/src/context/quiz-context.type.ts

```diff
@@ -0,0 +1,34 @@
+import { Dispatch } from "react";
+import { Option, Quiz } from "../database/data.type";
+
+export type VALUE_TYPE = {
```

Comment:
- Try to have consistent naming for all the `type aliases` here you have used `ALL_UPPER_CASE_WITH_UNDERSCORES` and in other places you have used `UpperCamelCase`.
- Try to stick with one convention.
- I suggest you to go with second approach because first approach is generally used for constants.